setup.py

```diff
@@ -1,15 +1,80 @@
 #!/usr/bin/env python
+import os
+import subprocess
+import sys
+from glob import glob
+from setuptools import setup, find_packages, Extension
 
-try:
-    from setuptools import setup
-except ImportError:
-    from distribute_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup
+
+if os.path.exists('relic'):
+    sys.path.insert(1, 'relic')
+    import relic.release
+else:
+    try:
+        import relic.release
+    except ImportError:
+        try:
+            subprocess.check_call(['git', 'clone',
+                'https://github.com/jhunkeler/relic.git'])
+            sys.path.insert(1, 'relic')
+            import relic.release
+        except subprocess.CalledProcessError as e:
+            print(e)
+            exit(1)
+
+
+version = relic.release.get_info()
+relic.release.write_template(version, 'lib/reftools')
 
 setup(
-    setup_requires=['d2to1>=0.2.9', 'stsci.distutils>=0.3.2'],
-    d2to1=True,
-    use_2to3=False,
-    zip_safe=False
+    name = 'reftools',
+    version = version.pep386,
```

Comment:
1. Tagged 1.7 - `foo()` does not exist yet
2. 1.7.dev1 - `foo()` exists
3. 1.7.dev2 - `foo()` and `bar()` exist
4. 2.0 - `foo()` and `bar()` exist
5. 2.0.dev1 - `foo` no longer exists, `foo2` exists

`foo() ` belongs to <=2.0 and `foo2` belongs to >= 2.0.devXX

If a user installed the master branch copy on Jan 6 their version would be reported as (assuming no commits other occurred) as 2.0 and will still only have access to `foo()` and `bar()`, not `foo2()`